### PR TITLE
chore: add java-shared-config as parent for consolidated version management

### PR DESCRIPTION
**Pull request by mpeddada1**
Notable Changes:
- Removed surefire-junit47 provider which was conflicting with the native-maven-plugin's testing support. See https://github.com/googleapis/google-auth-library-java/pull/1329#discussion_r1388205323 for details. 
- Removed `fmt-maven-plugin ` to rely on the one defined in java-shared-config.
- Removed native profile to rely on the one defined in java-shared-config
- Removed `autovalue-java8` profile and the autovalue dependency declaration to use the one defined in java-shared-config. 
- Added `native-test` profile to explicitly include ITs and functional tests for native image compilation.

### CODE REVIEW

**Comment by mpeddada1** [comment]
From local testing, adding java-shared-config and keeping the profile that previously existed in google-auth-library-java continues to result in:
```
[ERROR] Failed to execute goal org.graalvm.buildtools:native-maven-plugin:0.9.28:test (test-native) on project google-auth-library-oauth2-http: Execution test-native of goal org.graalvm.buildtools:native-maven-plugin:0.9.28:test failed: Test configuration file wasn't found.
```

File: oauth2_http/pom.xml [hunk 1]
```diff
@@ -182,13 +144,6 @@
...
           <reportNameSuffix>sponge_log</reportNameSuffix>
         </configuration>
-        <dependencies>
-          <dependency>
-            <groupId>org.apache.maven.surefire</groupId>
-            <artifactId>surefire-junit47</artifactId>
-            <version>3.0.0-M5</version>
-          </dependency>
-        </dependencies>
       </plugin>
```
Comment (mpeddada1) [inline oauth2_http/pom.xml:147]: A minimum of junit 5.8.1 is required for tests to run with native image compilation([reference](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#testing-support)) and this dependency prevents tests from running in junit 5. Removing it resolves the following error:
```
Execution test-native of goal org.graalvm.buildtools:native-maven-plugin:0.9.23:test failed: Test configuration file wasn't found.
```

According to the [maven documentation](https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit.html#provider-selection), the provider is automatically selected based on the junit version present in the project dependencies.

**Comment by mpeddada1** [comment]
Update: The graalvm checks are passing but the `ci/lint` check is now failing with:
```
Error:  Failed to execute goal com.coveo:fmt-maven-plugin:2.13:check (default-cli) on project google-auth-library-credentials: Execution default-cli of goal com.coveo:fmt-maven-plugin:2.13:check failed: An API incompatibility was encountered while executing com.coveo:fmt-maven-plugin:2.13:check: java.lang.NoSuchMethodError: 'java.lang.String com.google.googlejavaformat.java.ImportOrderer.reorderImports(java.lang.String, com.google.googlejavaformat.java.JavaFormatterOptions$Style)'
```

File: pom.xml [hunk 2]
```diff
@@ -430,49 +426,6 @@
         </plugins>
       </build>
     </profile>
-    <profile>
```
Comment (mpeddada1) [inline pom.xml:433]: Rely on autovalue-java8 profile defined in https://github.com/googleapis/java-shared-config/blob/cd05f0cad6fafc470d095c6ac428b87282c219fc/pom.xml#L734-L777

File: pom.xml [hunk 3]
```diff
@@ -209,14 +213,6 @@
           <artifactId>maven-dependency-plugin</artifactId>
           <version>3.6.1</version>
         </plugin>
-        <plugin>
```
Comment (mpeddada1) [inline pom.xml:212]: Rely on fmt-maven-plugin define in java-shared-config: https://github.com/googleapis/java-shared-config/blob/cd05f0cad6fafc470d095c6ac428b87282c219fc/pom.xml#L221-L235

File: oauth2_http/pom.xml [hunk 4]
```diff
@@ -23,57 +23,19 @@
...
       <build>
         <plugins>
           <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-surefire-plugin</artifactId>
-            <!-- Must use older version of surefire plugin for native-image testing. -->
-            <version>2.22.2</version>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>3.0.0-M5</version>
```
Comment (burkedavison) [inline oauth2_http/pom.xml:32]: Can we use the version defined in java-shared-config? Currently `3.2.1`: https://github.com/googleapis/java-shared-config/blob/main/pom.xml#L67
```suggestion
          <version>${surefire.version}</version>
```

**Review by burkedavison — APPROVED** [review]

File: oauth2_http/pom.xml (hunk 4 above)
Comment (mpeddada1) [reply oauth2_http/pom.xml]: Thanks! Done.

**Pull request merged**